src/distributors/base-installer.ts

```diff
@@ -0,0 +1,118 @@
+import * as tc from '@actions/tool-cache';
+import * as core from '@actions/core';
+import semver from 'semver';
+import path from 'path';
+import * as httpm from '@actions/http-client';
+import { getVersionFromToolcachePath } from '../util';
+
+export interface JavaInstallerOptions {
+    version: string;
+    arch: string;
+    javaPackage: string;
+}
+
+export interface JavaInstallerResults {
+    javaVersion: string;
+    javaPath: string;
+}
+
+export interface JavaDownloadRelease {
+    resolvedVersion: string;
+    link: string;
+}
```

Comment:
Consider moving these interfaces to `base-models.ts`

Reply:
Done. Thank you.